Data/models/DataSaved.swift

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+
+public final class DataSaved: NSManagedObject, CRUD {
+    @NSManaged public var savedUrl: String
+    @NSManaged public var amount: String
+    
+    public class func get(with savedUrl: String) -> DataSaved? {
+        let predicate = NSPredicate(format: "\(#keyPath(DataSaved.savedUrl)) == %@", savedUrl)
+        return first(where: predicate)
+    }
+    
+    public class func all() -> [DataSaved] {
+        all() ?? []
+    }
+    
+    public class func delete(with savedUrl: String) {
+        let context = DataController.viewContext
+        
+        if let item = get(with: savedUrl) {
+            item.delete(context: .existing(context))
+            if context.hasChanges {
+                try? context.save()
+            }
+        }
+    }
+    
+    public class func insert(savedUrl: String, amount: String) {
+        let context = DataController.viewContext
+        
+        guard let entity =  NSEntityDescription.entity(forEntityName: "DataSaved", in: context) else {
+            return
+        }
+        
+        let source = DataSaved(entity: entity, insertInto: context)
```

Comment:
See my other comment about avoiding writing on main thread

Reply:
Yeah you are right, it can be totally done in background thread here actually. The Data Saved Model doesnt have the urgency to save it to context immediately.

Reply:
Usually we do writes on main thread only if the view must respond model changes immediately and fetch results controller is not implemented on such screen

Reply:
Moved to background thread.